resources/web/wwi/protoVisualizer/Parameter.js

```diff
@@ -28,6 +38,15 @@ export default class Parameter {
     if (v.type() !== this.type)
       throw new Error('Type mismatch, setting ' + stringifyType(v.type()) + ' to ' + stringifyType(this.type) + ' parameter.');
 
+    if (this.restrictions.length > 0) {
+      let isValueAcceptable = false;
+      for (const item of this.restrictions)
+        isValueAcceptable = isValueAcceptable || (v instanceof SFNode && v.value === null) || item.equals(v);
```

Comment:
Wouldn't it be better to break from the loop as soon as `isValueAcceptable` is set to true or maybe use a Set to completely avoid the loop?

Reply:
The loop can indeed be improved, but using a Set might be overkill since in practice we usually have at most a couple restrictions